plugin/src/main/java/org/opensearch/sql/plugin/SQLPlugin.java

```diff
@@ -277,13 +277,23 @@ public ScheduledJobParser getJobParser() {
 
   @Override
   public List<ExecutorBuilder<?>> getExecutorBuilders(Settings settings) {
-    return singletonList(
+    // The worker pool is the primary pool where most of the work is done. The background thread
+    // pool is a separate queue for asynchronous requests to other nodes. We keep them separate to
+    // prevent deadlocks during async fetches on small node counts. Tasks in the background pool
+    // should do no work except I/O to other services.
+    return List.of(
         new FixedExecutorBuilder(
             settings,
-            AsyncRestExecutor.SQL_WORKER_THREAD_POOL_NAME,
+            SQL_WORKER_THREAD_POOL_NAME,
             OpenSearchExecutors.allocatedProcessors(settings),
             1000,
-            null));
```

Comment:
This previous null prefix meant that we were literally registering the global YML setting `null` to configure the SQL thread pool:

```yml
null:
  size: 30
```

I updated this to use the proper thread_pool setting prefix with accurate naming. This also meant tweaking the thread pool to be named with an underscore, which should be a safe change.

Reply:
unfortunate.